pyfive/dataobjects.py

```diff
@@ -321,7 +321,8 @@ def compression_opts(self):
         if GZIP_DEFLATE_FILTER in self._filter_ids:
             gzip_entry = [d for d in self.filter_pipeline
                           if d['filter_id'] == GZIP_DEFLATE_FILTER][0]
-            return gzip_entry['client_data'][0]
+            key = {0:'client_data_values',1:'client_data'}['client_data' in gzip_entry]
+            return gzip_entry[key][0]
         return None
```

Comment:
By the documentation pointed to in your issue, [[link](https://davis.lbl.gov/Manuals/HDF5-1.8.7/H5.format.html#FilterMessage)], it seems there is only a single list of client data values. Instead of your suggested fix, I think we instead should search-and-replace in the code and never produce any dictionary with `"client_data"` as the key. (disclaimer: I haven't actually studied the source code yet)

Reply:
I can see arguments for doing it both ways, the code I've committed is as faithful to what is in the file at point of use, to do different would require hacking the message structure before using it ... I'd defer to the package owners on what is the best option.

Reply:
That's fine, thanks for elaborating!

Reply:
(At some point you'd have to do what I've got above, insofar as we see files with both compression option structures, we can't control what we will get to have to read.)